tlsconfig/certpool_go17.go

```diff
@@ -0,0 +1,15 @@
+// +build go1.7
+
+package tlsconfig
+
+import "crypto/x509"
+
+// SystemCertPool returns a copy of the system cert pool,
+// returns an empty cert pool if the system pool fails to load.
+func SystemCertPool() *x509.CertPool {
+	certpool, err := x509.SystemCertPool()
+	if err != nil {
+		return x509.NewCertPool()
```

Comment:
We can check for the windows ourselves here to avoid calling this when using windows, see https://golang.org/src/crypto/x509/cert_pool.go?s=730:770#L22. In that case we would still end up calling `NewCertPool` for windows.